// codblo/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/code.db');
var udb = new Database(__dirname+'/db/usr.db');

// === log ===
exports.typBlo=function(typ){
var stm=db.prepare('select *,rowid from linux where type=? order by rowid desc');
try{var row=stm.all(typ);
}catch(err){console.log(err);}
return row;
};

exports.allBlo=function(){
var stm=db.prepare('select * from linux');
var row=stm.all();
return row;
};

exports.terBlo=function(ter){
var stm=db.prepare('select * from linux where term=?');
try{var row=stm.get(ter);
}catch(err){console.log(err);}
return row;
};

exports.titBlo=function(tit){
var stm=db.prepare('select *,rowid from linux where title=?');
try{var row=stm.get(tit);
}catch(err){console.log(err);}
return row;
};

exports.idBlo=function(id){
var stm=db.prepare('select *,rowid from linux where rowid=?');
try{var row=stm.get(id);
}catch(err){console.log(err);}
return row;
};
// === run

// ins
exports.insBlo=function(typ,ter,tit,lin,bod){
var stm=db.prepare('insert into linux values(?,?,?,?,?)')
try{var row=stm.run(typ,ter,tit,lin,bod)
}catch(err){console.log(err);}
};

//upd
exports.upTyp=function(type,id){
var stm=db.prepare('update linux set type=? where rowid=? ')
try{stm.run(type,id)
}catch(err){console.log(err);}
};

exports.upTer=function(term,id){
var stm=db.prepare('update linux set term=? where rowid=? ')
try{stm.run(term,id)
}catch(err){console.log(err);}
};

exports.upTit=function(title,id){
var stm=db.prepare('update linux set title=? where rowid=? ')
try{stm.run(title,id)
}catch(err){console.log(err);}
};

exports.upLin=function(lin,id){
var stm=db.prepare('update linux set link=? where rowid=? ')
try{stm.run(lin,id)
}catch(err){console.log(err);}
};

exports.upBod=function(bod,id){
var stm=db.prepare('update linux set body=? where rowid=? ')
try{stm.run(bod,id)
}catch(err){console.log(err);}
};

// usr
exports.getSid=function(usr,pss){
var stm=udb.prepare('select *,rowid from usr where usr=? and pss=?')
try{var row=stm.get(usr,pss);
}catch(err){console.log(err);}
return row;
};

exports.getUsr=function(usr,pss){
var stm=udb.prepare('select *,rowid from usr where usr=? and pss=?')
try{var row=stm.get(usr,pss);
}catch(err){console.log(err);}
return row;
};

exports.upSid=function(sid,usr,pss){
var stm=udb.prepare('update usr set sid=? where usr=? and pss=? ')
try{stm.run(sid,usr,pss)
}catch(err){console.log(err);}
};
